src/test/e2e/api/admin/api-token.filtering.auth.e2e.test.ts

```diff
@@ -0,0 +1,126 @@
+import {
+    type IUnleashTest,
+    setupAppWithAuth,
+} from '../../helpers/test-helper.js';
+import dbInit, { type ITestDb } from '../../helpers/database-init.js';
+import getLogger from '../../../fixtures/no-logger.js';
+import { ApiTokenType } from '../../../../lib/types/model.js';
+import { DEFAULT_ENV } from '../../../../lib/util/index.js';
+import {
+    SYSTEM_USER_AUDIT,
+    SYSTEM_USER_ID,
+} from '../../../../lib/server-impl.js';
+
+let db: ITestDb;
+let app: IUnleashTest;
+
+beforeAll(async () => {
+    db = await dbInit('token_api_project_token_filtering', getLogger);
+    app = await setupAppWithAuth(
+        db.stores,
+        {
+            experimental: {
+                flags: {
+                    strictSchemaValidation: true,
+                },
+            },
+        },
+        db.rawDatabase,
+    );
+});
+
+afterAll(async () => {
+    if (db) {
+        await db.destroy();
+    }
+    await app.destroy();
+});
+
+afterEach(async () => {
+    await db.stores.apiTokenStore.deleteAll();
+});
+
+test('finds tokens and stuff', async () => {
```

Comment:
This killed me :laughing: 

Can we pick a better name though? I tend to use tests to try and understand what logic is being enforced by them so what about 'filters out tokens unavailable to user'?